### PR TITLE
refactor(webserver): web server refactor

### PR DESCRIPTION
**Pull request by vinayg-intel**
## PR Checklist

- [ ] Unit Tests have been added for new changes
- [ ] API tests have been updated if applicable
- [ ] All commented code has been removed
- [ ] If you've added a dependency, you've ensured license is compatible with Apache 2.0 and clearly outlined the added dependency.

### CODE REVIEW

File: src/server/webserver.ts [hunk 1]
```diff
@@ -65,192 +65,17 @@ export class WebServer {
       this.app.use(cors())
 
       // Handles the Bad JSON exceptions
-      this.app.use(parser.json(), (err, req, res, next) => {
+      this.app.use(parser.json(), (err: any, req: Request, res: any, next: () => void) => {
```
Comment (rsdmike) [inline src/server/webserver.ts:68]: use Express.Response

File: src/server/webserver.ts [hunk 2]
```diff
@@ -267,11 +92,9 @@ export class WebServer {
       }, routes)
 
       // Handle upgrade on websocket
-      this.server.on('upgrade', (request, socket, head) => {
+      this.server.on('upgrade', (request: IncomingMessage, socket: Socket, head: Buffer) => {
```
Comment (rsdmike) [inline src/server/webserver.ts:95]: remove amtFactory

File: src/test/webserver.test.ts [hunk 3]
```diff
@@ -0,0 +1,44 @@
+import * as fs from 'fs'
```
Comment (rsdmike) [inline src/test/webserver.test.ts:1]: tests need to be written

File: src/utils/httpInterceptor.ts [hunk 4]
```diff
@@ -0,0 +1,209 @@
...
+ * Copyright (c) Intel Corporation 2021
+ * SPDX-License-Identifier: Apache-2.0
+ **********************************************************************/
+
+import { AmtMode, Args, Connection, ConnectionType } from '../models/models'
+import Common from './common'
+
+export const HttpInterceptorAuthentications: {[key: string]: string} = {}
+
+export class HttpInterceptor {
```
Comment (rsdmike) [inline src/utils/httpInterceptor.ts:11]: not sure this is needed anymore --- can we use HttpHandler?

File: src/utils/wsRedirect.ts [hunk 5]
```diff
@@ -0,0 +1,224 @@
+import { connect } from 'tls'
```
Comment (rsdmike) [inline src/utils/wsRedirect.ts:1]: needs more test -- currently at 60%

**Review by rsdmike — CHANGES_REQUESTED** [review]

File: src/test/webserver.test.ts (hunk 3 above)
Comment (vinayg-intel) [reply src/test/webserver.test.ts]: updated tests

File: src/utils/wsRedirect.ts (hunk 5 above)
Comment (vinayg-intel) [reply src/utils/wsRedirect.ts:1]: updated tests

File: src/test/webserver.test.ts (hunk 3 above)
Comment (vinayg-intel) [reply src/test/webserver.test.ts]: test updated

File: src/server/webserver.ts (hunk 2 above)
Comment (vinayg-intel) [reply src/server/webserver.ts:84]: removed

File: src/server/webserver.ts (hunk 1 above)
Comment (vinayg-intel) [reply src/server/webserver.ts]: updated

File: src/utils/wsRedirect.ts (hunk 5 above)
Comment (vinayg-intel) [reply src/utils/wsRedirect.ts:1]: test updated

**Review by rsdmike — CHANGES_REQUESTED** [review]
KVM doesn't seem to work for me. 
1) Get AMT Features isn't returning the correct value of consent of 'none', its missing. This then assumes the device is in CCM and tries to request a user consent code. 
2) User consent code request comes back with NOT_READY
3) Addressing the above issue --- MPS crashes with:
```
2022-01-12T15:29:50.862Z debug: Opening web socket connection to dc09fce2-a602-11ea-90a8-90cf9ac8ee00: 16994.
2022-01-12T15:29:50.862Z verbose: getting secrets from path: secret/data/devices/dc09fce2-a602-11ea-90a8-90cf9ac8ee00
2022-01-12T15:29:50.867Z debug: got data back from vault at path: secret/data/devices/dc09fce2-a602-11ea-90a8-90cf9ac8ee00
2022-01-12T15:29:50.868Z debug: Creating credential
2022-01-12T15:29:50.891Z error: uncaughtException: ws.forwardclient.write is not a function
TypeError: ws.forwardclient.write is not a function
    at WsRedirect.handleMessage (d:\Development\Intel\ActiveEdge\open-amt-cloud-toolkit\mps\dist\utils\wsRedirect.js:71:26)
    at WebSocket.<anonymous> (d:\Development\Intel\ActiveEdge\open-amt-cloud-toolkit\mps\dist\utils\wsRedirect.js:27:22)
    at WebSocket.emit (node:events:390:28)
    at Receiver.receiverOnMessage (d:\Development\Intel\ActiveEdge\open-amt-cloud-toolkit\mps\node_modules\ws\lib\websocket.js:1098:20)
    at Receiver.emit (node:events:390:28)
    at Receiver.dataMessage (d:\Development\Intel\ActiveEdge\open-amt-cloud-toolkit\mps\node_modules\ws\lib\receiver.js:513:14)
    at Receiver.getData (d:\Development\Intel\ActiveEdge\open-amt-cloud-toolkit\mps\node_modules\ws\lib\receiver.js:446:17)
    at Receiver.startLoop (d:\Development\Intel\ActiveEdge\open-amt-cloud-toolkit\mps\node_modules\ws\lib\receiver.js:148:22)
    at Receiver._write (d:\Development\Intel\ActiveEdge\open-amt-cloud-toolkit\mps\node_modules\ws\lib\receiver.js:83:10)
    at writeOrBuffer (node:internal/streams/writable:389:12)
2022-01-12T15:29:50.892Z debug: signal received :
2022-01-12T15:29:50.933Z verbose: executed query: {"text":"\n      UPDATE devices \n      SET mpsinstance=$2, connectionstatus=$3 \n      WHERE mpsinstance=$1 and tenantId = $4","duration":41,"rows":2}
2022-01-12T15:29:50.934Z debug: Clean DB instance before exit
2022-01-12T15:29:51.940Z debug: signal received :
```

File: src/utils/httpInterceptor.ts (hunk 4 above)
Comment (brianosburn-intel) [reply src/utils/httpInterceptor.ts:11]: I dont believe httphandler will be a replacement. I tried parsing the messages with httpZ.parse(this.amt.acc) as HttpZResponseModel and wasn't able to get any messages to parse without throwing an exception.

**Review by rsdmike — APPROVED** [review]

**Pull request merged**